opensoundscape/audio.py

```diff
@@ -349,39 +339,48 @@ def split_and_save(
     clip_duration,
     clip_overlap=0,
     final_clip=None,
+    raven_file=None,
     dry_run=False,
 ):
     """ Split audio into clips and save them to a folder
 
     Args:
-        audio:          The input Audio to split
-        destination:    A folder to write clips to
-        prefix:         A name to prepend to the written clips
-        clip_duration:  The duration of each clip in seconds
-        clip_overlap:   The overlap of each clip in seconds [default: 0]
-        final_clip:     Possible options (any other input will ignore the final clip entirely) [default: None]
-                            - "remainder":          Include the remainder of the Audio
-                                                        (clip will not have clip_duration length)
-                            - "full":               Increase the overlap to yield a clip with clip_duration
-                            - "extend":             Similar to remainder but extend the clip to clip_duration
-        dry_run:        If True, skip writing audio and just return clip DataFrame [default: False]
-    
+        audio:              The input Audio to split
+        destination:        A folder to write clips to
+        prefix:             A name to prepend to the written clips
+        clip_duration:      The duration of each clip in seconds
+        clip_overlap:       The overlap of each clip in seconds [default: 0]
+        final_clip (str):   Behavior if final_clip is less than clip_duration seconds long. [default: None]
+            By default, ignores final clip entirely.
+            Possible options (any other input will ignore the final clip entirely),
+                - "remainder":  Include the remainder of the Audio (clip will not have clip_duration length)
+                - "full":       Increase the overlap to yield a clip with clip_duration length
+                - "extend":     Similar to remainder but extend (repeat) the clip to reach clip_duration length
+        dry_run (bool):      If True, skip writing audio and just return clip DataFrame [default: False]
+
     Returns:
         pandas.DataFrame containing begin and end times for each clip from the source audio
     """
 
     clips = audio.split(
         clip_duration=clip_duration, clip_overlap=clip_overlap, final_clip=final_clip
     )
+    clip_names = []
     for clip in clips:
         clip_name = (
             f"{destination}/{prefix}_{clip['begin_time']}s_{clip['end_time']}s.wav"
         )
-        clip["filename"] = clip_name
+        clip_names.append(clip_name)
         if not dry_run:
             clip["clip"].save(clip_name)
 
     # Convert [{k: v}] -> {k: [v]}
-    return pd.DataFrame(
-        {key: [clip[key] for clip in clips] for key in clips[0].keys() if key != "clip"}
+    sels = pd.DataFrame(
```

Comment:
what is "sels"?

Reply:
I think "sels" was supposed to mean "selections." The return is identical to the return that was already there, just formatted by naming the variable first. I'm not sure why I did this, though, and the name doesn't make sense, so I'll change it.

Reply:
Fixed this